### ISSUE TITLE
@babel/preset-env and the term useBuiltIns

### ISSUE THREAD

**Issue by thernstig** [babel/babel#13156]
## Feature Request

**Is your feature request related to a problem?**
Having recently learned many ins and outs of `@babel/preset-env` the term `useBuiltIns` has been one of the harder ones to grok. See https://babeljs.io/docs/en/babel-preset-env#usebuiltins

**Describe the solution you'd like**
I wonder if there is not a better term than `useBuiltIns` for what it does. It almost exclusively used for updating/adding polyfills (with the exception of that it might also add `regeneratorRuntime`).

Why could it not be renamed to `polyfills`? Or, what is even the reason for the term `useBuiltIns`? It is certainly not telling us to use the JavaSscript built-in global vars, or built-in static and instance properties. It is doing the opposite of this.

**Documentation, Adoption, Migration Strategy**
Would need to be done in babel 8 as it is a breaking change. Also maybe it is possible that if it is renamed to `polyfills` then maybe the handling of `regeneratorRuntime` (which is currently  added by `useBuiltIns: 'usage'` should be a separate option.

**Comment by nicolo-ribaudo** [comment]
To add to the confusion, we have another `useBuiltIns` option in a plugin that does a different thing :sweat_smile:  

However, you can already be explicit by configuring Babel like this:
```js
{
  "targets": "> 2%",
  "presets": ["@babel/preset-env"],
  "plugins": [
    ["polyfill-corejs3", { "method": "usage-global" }]
  ]
}
```

This has a few advantages:
- It makes it clear that it's about polyfills
- You don't have to worry about "why is there `corejs` both in `preset-env` and in `transform-runtime`?"
- Authors of other polyfills can provide their custom Babel plugins, using a [standard interface](https://github.com/babel/babel-polyfills)

This will be the recommended polyfills injection method in the future, but you can already use it now (and it's what `@babel/preset-env` uses internally).

**Comment by thernstig** [comment]
@nicolo-ribaudo I assume that also adds `regeneratorRuntime` (which is not a polyfill per se) where needed?

Also, how is that combined with https://babeljs.io/docs/en/babel-plugin-transform-runtime ?

**Comment by nicolo-ribaudo** [comment]
> Also, how is that combined with babeljs.io/docs/en/babel-plugin-transform-runtime ? Reading the babel/babel-polyfills it seems it is not needed anymore?

We want to keep `babel-plugin-transform-runtime` only for Babel helpers, and delegate polyfills to external plugins.

Currently we have one package (`transform-runtime`) that does two things (helpers + polyfills), and two packages (`transform-runtime` and `preset-env`) that both inject polyfills but in different ways.

We think that having one package for each thing (one for helpers, one for each polyfill, and one for transforms) will be less confusing.

> why isn't that plugin and setup default for `@babel/preset-env`?

Because there isn't an official "Babel polyfill": they are all third-party polyfills, so we want to let users choose between different alternatives.

> But would I not need to use es-shims and regenerator-runtime plugins as well to be completely compliant with my targets?

`es-shims` is just an alternative to `core-js`: they are both polyfills (well, `es-shims` is more like _a set_ of polyfills).
The current `regenerator-runtime` situation is currently a bit confusing because, even if it's technically handled as a polyfill, we should consider ti exactly like we consider all the other Babel runtime helpers.

However, yes: if you compile generators and you don't use `transform-runtime` (which handles `regenerator-runtime` exactly how it handles the other helpers) you'll want to include `regenerator-runtime`.

**Comment by thernstig** [comment]
> We want to keep babel-plugin-transform-runtime only for Babel helpers, and delegate polyfills to external plugins.

Is that true of today, if I use it?

> es-shims is just an alternative to core-js: they are both polyfills (well, es-shims is more like a set of polyfills).

Ok, I never found this clear anywhere that they were alternatives. Maybe it should be explicit somewhere in the new docs. But how come then https://babeljs.io/docs/en/babel-plugin-transform-runtime#corejs has that `corejs` option but no `es-shims` option. It sounds as if using `es-shims`, transform-runtime can not handle that as well. I'm thus very confused as to which those two alternatives I should use.

> However, yes: if you compile generators and you don't use transform-runtime (which handles regenerator-runtime exactly how it handles the other helpers) you'll want to include regenerator-runtime.

So if I have this, I should be fine? It will give me transforms, polyfills (with corejs3), regenerator-runtime and also convert all code to use import for the helpers and runtime?

```json
{
  "targets": "> 2%",
  "presets": ["@babel/preset-env"],
  "plugins": [
    ["polyfill-corejs3", { "method": "usage-global" }],
    [
      "@babel/plugin-transform-runtime",
      {
        "corejs": 3,
      }
    ]
  ]
}
```

**Comment by thernstig** [comment]
@nicolo-ribaudo  I just found and read https://github.com/babel/babel-polyfills/blob/main/docs/migration.md - reading that page it sounds like I should do:

```js
{
  "targets": { "firefox": 42 },
  "presets": ["@babel/preset-env"],
  "plugins": [
    ["polyfill-corejs3", {
      "method": "usage-global"
    }],
    ["@babel/transform-runtime", {
      "corejs": 3
    }]
  ]
}
```

This would use corejs 3, but also with the help of `transform-runtime` minimize the build bundle. Either that, or I cannot grok what https://babeljs.io/docs/en/babel-plugin-transform-runtime#corejs actually does. Because my code proposal here does not fit with the migration guides third `core-js@3` _New configuration_. But that one shows `usage-pure` which does not sound like something I want as it is not a library author as the docs mention it is best used for. I also inverted the plugins order as I assume I want `transform-runtime` to always run last?
